examples/jsm/effects/ParallaxBarrierEffect.js

```diff
@@ -11,102 +11,106 @@ import {
 
-	var _stereo = new StereoCamera();
+		const _scene = new Scene();
 
-	var _params = { minFilter: LinearFilter, magFilter: NearestFilter, format: RGBAFormat };
+		const _stereo = new StereoCamera();
 
-	var _renderTargetL = new WebGLRenderTarget( 512, 512, _params );
-	var _renderTargetR = new WebGLRenderTarget( 512, 512, _params );
+		const _params = { minFilter: LinearFilter, magFilter: NearestFilter, format: RGBAFormat };
 
-	var _material = new ShaderMaterial( {
+		const _renderTargetL = new WebGLRenderTarget( 512, 512, _params );
+		const _renderTargetR = new WebGLRenderTarget( 512, 512, _params );
 
-		uniforms: {
+		const _material = new ShaderMaterial( {
 
-			'mapLeft': { value: _renderTargetL.texture },
-			'mapRight': { value: _renderTargetR.texture }
+			uniforms: {
 
-		},
+				'mapLeft': { value: _renderTargetL.texture },
+				'mapRight': { value: _renderTargetR.texture }
 
-		vertexShader: [
+			},
 
-			'varying vec2 vUv;',
+			vertexShader: [
 
-			'void main() {',
+				'varying vec2 vUv;',
 
-			'	vUv = vec2( uv.x, uv.y );',
-			'	gl_Position = projectionMatrix * modelViewMatrix * vec4( position, 1.0 );',
+				'void main() {',
 
-			'}'
+				'	vUv = vec2( uv.x, uv.y );',
+				'	gl_Position = projectionMatrix * modelViewMatrix * vec4( position, 1.0 );',
 
-		].join( '\n' ),
+				'}'
 
-		fragmentShader: [
+			].join( '\n' ),
 
-			'uniform sampler2D mapLeft;',
-			'uniform sampler2D mapRight;',
-			'varying vec2 vUv;',
+			fragmentShader: [
 
-			'void main() {',
+				'uniform sampler2D mapLeft;',
+				'uniform sampler2D mapRight;',
+				'varying vec2 vUv;',
 
-			'	vec2 uv = vUv;',
+				'void main() {',
 
-			'	if ( ( mod( gl_FragCoord.y, 2.0 ) ) > 1.00 ) {',
+				'	vec2 uv = vUv;',
 
-			'		gl_FragColor = texture2D( mapLeft, uv );',
+				'	if ( ( mod( gl_FragCoord.y, 2.0 ) ) > 1.00 ) {',
 
-			'	} else {',
+				'		gl_FragColor = texture2D( mapLeft, uv );',
 
-			'		gl_FragColor = texture2D( mapRight, uv );',
+				'	} else {',
 
-			'	}',
+				'		gl_FragColor = texture2D( mapRight, uv );',
 
-			'}'
+				'	}',
 
-		].join( '\n' )
+				'}'
 
-	} );
+			].join( '\n' )
 
-	var mesh = new Mesh( new PlaneGeometry( 2, 2 ), _material );
-	_scene.add( mesh );
+		} );
 
-	this.setSize = function ( width, height ) {
+		const mesh = new Mesh( new PlaneGeometry( 2, 2 ), _material );
+		_scene.add( mesh );
 
-		renderer.setSize( width, height );
+		this.setSize = function ( width, height ) {
 
-		var pixelRatio = renderer.getPixelRatio();
+			renderer.setSize( width, height );
 
-		_renderTargetL.setSize( width * pixelRatio, height * pixelRatio );
-		_renderTargetR.setSize( width * pixelRatio, height * pixelRatio );
+			const pixelRatio = renderer.getPixelRatio();
 
-	};
+			_renderTargetL.setSize( width * pixelRatio, height * pixelRatio );
+			_renderTargetR.setSize( width * pixelRatio, height * pixelRatio );
 
-	this.render = function ( scene, camera ) {
+		};
 
-		scene.updateMatrixWorld();
+		this.render = function ( scene, camera ) {
```

Comment:
The internal properties could be defined outside of the class like [here](https://github.com/mrdoob/three.js/blob/eeaa508e1e67c87951684c35c7fbb965dd4e1aca/examples/jsm/animation/CCDIKSolver.js#L16-L26), `scope` can be replaced just by `this`, and `render` and `setSize` could become class methods, no?

Reply:
I'm not sure this kind of refactoring is wanted. It requires that a lot of variables are added to the public namespace so they are accessible by all methods.

If the current code structure does not interfere with tree-shaking, I would like to avoid changing it (and just develop new classes differently).

Reply:
>  It requires that a lot of variables are added to the public namespace

Public namespace? ES modules are locally scoped 😇

Reply:
I mean variables are defined in the constructors scope. It would be necessary to add them to `this`.

Reply:
@marcofugaro Don't mess with Mugen's flow! 😁 Feel free to do a second pass later 😉

Reply:
okay! didn't mean to criticize Mugen's work in any way 😄 those ES6 PRs are awesome! keep it up!

Reply:
I'm totally fine when this is revisited at a later point. @marcofugaro Feel free to tackle this if you like 😊 .